src/vaults/ProofsFetcher.ts

```diff
@@ -0,0 +1,29 @@
+import { getAllProofs, Proofs } from "../utils/getCurrentOnChainDistribution";
+import { EpochConfig } from "../ages";
+import { getEpochFromId } from "../utils/timestampToEpoch";
+
+export interface ProofsFetcherInterface {
+  fetchProofs: (address: string, epochToId?: string) => Promise<Proofs[]>;
+  getEpochFromId: (epochId: string) => EpochConfig;
+}
+
+export default class ProofsFetcher implements ProofsFetcherInterface {
+  async fetchProofs(address: string, epochToId?: string): Promise<Proofs[]> {
+    const proofs = getAllProofs()
+      .reverse()
+      .filter((proofs) => !!proofs.proofs[address.toLowerCase()]?.amount);
+    if (epochToId) {
+      const epochConfig = getEpochFromId(epochToId);
+      if (!epochConfig) throw Error(`Invalid epoch id ${epochToId}`);
```

Comment:
Is this necessary? Isn't it enough to look for an `proof.epoch === epochToId`?

Reply:
Youre right but the problem is not the same. For example, if any user miss name an epoch (ex: age-1-epoch-1 instead of age1-epoch1) If I remove this check, It will have the error message No MORPHO distributed for the vault ${address} in epoch ${epochToId}
Where this message is suited for valid epoch but no rewards distributed to the vault

Reply:
Nice